aws/resource_aws_opsworks_chef.go

```diff
@@ -0,0 +1,335 @@
+package aws
+
+import (
+	"log"
+	"strings"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/service/opsworkscm"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+const EngineModel = "Single"
+const EngineVersion = "12"
+
+func resourceAwsOpsworksChef() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsOpsworksChefCreate,
+		Read:   resourceAwsOpsworksChefRead,
+		Update: resourceAwsOpsworksChefUpdate,
+		Delete: resourceAwsOpsworksChefDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			// TODO: do we want this? I think this is another "magic" thing that happens
+			// I think we can still associate a public IP later if we want? :shrug:
+			"associate_public_ip_address": {
+				Type:     schema.TypeBool,
+				Optional: true,
+				Default:  false,
+			},
+
+			"backup_retention_count": {
+				Type:     schema.TypeInt,
+				Optional: true,
+				Default:  1,
+			},
+
+			"disable_automated_backup": {
+				Type:     schema.TypeBool,
+				Optional: true,
+				Default:  false,
+			},
+
+			// TODO:
+			// I kinda want to make this required because how else are we going to get at that info?
+			// Also, since these are write-only and not returned by the API, we'll probably need a custom diff function that ignores them?
+			"chef_pivotal_key": {
+				Type:      schema.TypeString,
+				Optional:  true,
+				Default:   nil,
+				Sensitive: true,
+			},
+
+			"chef_delivery_admin_password": {
+				Type:      schema.TypeString,
+				Optional:  true,
+				Default:   nil,
+				Sensitive: true,
+			},
+
+			// TODO:
+			// default and only valid. I'll leave it configurable for future proofing but
+			// validation failure, maybe?
```

Comment:
These are required parameters to the API call, but they only have one valid value at the moment.

Should I remove these from the resource and hardcode the values, leave them on the resource but enforce them in validation, or leave them on the resource, make them optional with these values as the default, and let the API handle validation? Or something else entirely?